virt_lightning/__init__.py

```diff
@@ -197,40 +198,42 @@ def kvm_binary(self):
             raise Exception("Failed to find the kvm binary in: ", paths)
 
     def init_network(self, bridge_name):
-        if self.conn.getURI().startswith("qemu:///session"):
-            try:
-                bridge_netiface = netifaces.ifaddresses(bridge_name)
-            except ValueError:
-                print("Bridge not found:", bridge_name)
-                exit(1)
-            ipconfig = bridge_netiface[netifaces.AF_INET]
-            self.gateway = ipaddress.IPv4Interface(
-                "{addr}/{netmask}".format(**ipconfig[0])
-            )
-            self.dns = self.gateway
-            self.network = self.gateway.network
-        elif self.conn.getURI().startswith("qemu:///system"):
-            print("system")
+        try:
+            bridge_netiface = netifaces.ifaddresses(bridge_name)
+        except ValueError:
+            print("Bridge not found:", bridge_name)
+            exit(1)
+        ipconfig = bridge_netiface[netifaces.AF_INET]
+        self.gateway = ipaddress.IPv4Interface("{addr}/{netmask}".format(**ipconfig[0]))
+        self.dns = self.gateway
+        self.network = self.gateway.network
 
     def init_storage_pool(self, storage_pool):
         try:
             self.storage_pool_obj = self.conn.storagePoolLookupByName(storage_pool)
             if not self.get_storage_dir().is_dir():
                 raise Exception("Missing storage directory:", self.get_storage_dir())
+            if not self.storage_pool_obj.isActive():
+                self.storage_pool_obj.create(0)
             return
         except libvirt.libvirtError as e:
             if e.get_error_code() == libvirt.VIR_ERR_NO_STORAGE_POOL:
                 print("Storage pool is missing.")
                 pass
 
-        if os.geteuid() == 0:
+        if self.conn.getURI().startswith("qemu:///system"):
             storage_dir = pathlib.PosixPath("/var/lib/virt-lightning/pool")
-        else:
+        elif self.conn.getURI().startswith("qemu:///session"):
             storage_dir = pathlib.PosixPath("~/.local/share/virt-lightning/pool")
             storage_dir = storage_dir.expanduser()
+        else:
+            raise Exception("Unsupported libvirt URI")
 
         full_dir = storage_dir / "upstream"
-        full_dir.mkdir(parents=True, exist_ok=True)
+        try:
+            full_dir.mkdir(parents=True, exist_ok=True)
+        except PermissionError:
+            print(USER_CREATE_STORAGE_POOL_DIR.format(storage_dir=storage_dir))
```

Comment:
Handle all type exceptions here too.

Reply:
I don't want to catch all the potential exception. It's not a good practice, see:
https://stackoverflow.com/questions/21553327/why-is-except-pass-a-bad-programming-practice